webapp/certification/views.py

```diff
@@ -15,6 +15,36 @@
 )
 
 
+def certification_component_details(component_id):
+
+    try:
+        component = api.component_summary(component_id)
+    except requests.exceptions.HTTPError as error:
+        abort(error.response.status_code)
+
+    models_response = api.certified_models(
+        canonical_id__in=component["machine_canonical_ids"],
+        limit=0,
+    )
+
+    all_machines = models_response["objects"]
+
+    machines_by_id = {}
+    for machine in all_machines:
+        if machine["canonical_id"] not in machines_by_id:
```

Comment:
Do we get multiple machines with the same `canonical_id`?

Reply:
I didn't check every page obviously, but I left this because I thought this probably is an issue in the current site. But both Sheila and Robin are off, so not sure who to ask to double check this

Reply:
the certified_models call should only return one object for a machine. It's certified_model_details that would return more than one object for a machine if the machine has multiple certificates.

Reply:
Removed